### PR TITLE
Update runtime and scratch desktop

### PR DESCRIPTION
**Pull request by dylanmccall**
In addition to updating scratch-desktop and its ilk, along with the runtime (which all happened together due to requiring a newer version of NodeJS), I ended up doing _a few other things_ while I was here:

- I cleaned up a few details in `edu.mit.Scratch.yaml`, removing some old comments and unused code.
- The newer version of Electron means we can drop the `filesystem=home` permission.
-  I noticed [Turbowarp](https://github.com/TurboWarp/desktop) was handling `*.sb3` files from Scratch, so I added the same kind of support to the Scratch app. The mime type is a bit odd (`application/x.scratch.sb3` as opposed to `application/x-scratch-sb3`), but that appears to be the format Scratch upstream uses, as well as Turbowarp.

I ran into one strange issue where the URL for the scratch-gui git repo in `generated-sources.json` turned out to be a `git+ssh://` style URL, which Flathub wasn't impressed with. I'd love to explore why that happened, but for now, I manually changed it to an `https://` URL.

Also, note that I'm only updating to `org.freedesktop.Platform//22.08` for now. I'd like to bump all the way to 23.08 soon, but this was already tricky to line up (due to also needing a different version of NodeJS), so I think it's easier doing that separately.

I'm happy to split this all into different pull requests if you prefer.

### CODE REVIEW

**Comment by dylanmccall** [comment]
There's one last (I think) thing to do here, which is scratch-gui added a prepublish script. A prepublish script is yet another diabolical way a nodejs project can ruin the day of anyone who cares about reproducible builds. This one runs this glob of code: https://github.com/scratchfoundation/scratch-gui/blob/develop/scripts/prepublish.mjs#L113-L126. I was hoping to just ignore it and not download this file and call it a day, //but// it breaks the build, so i need to add some extra handling to the project's Python script that generates "asset-sources.json".

**Comment by refi64** [comment]
Thanks!

> I ran into one strange issue where the URL for the scratch-gui git repo in generated-sources.json turned out to be a git+ssh:// style URL, which Flathub wasn't impressed with. I'd love to explore why that happened, but for now, I manually changed it to an https:// URL.

Might be worth mentioning in the readme if that needs to be redone when flatpak-node-generator gets run?

> I was hoping to just ignore it and not download this file and call it a day, //but// it breaks the build, so i need to add some extra handling to the project's Python script that generates "asset-sources.json".

I don't see any changes there other than the URL update, was this no longer necessary or did they get accidentally left out of the commits?

**Comment by dylanmccall** [comment]
> Might be worth mentioning in the readme if that needs to be redone when flatpak-node-generator gets run?

Yeah, I can do that :) Maybe I'll add a note to run `flatpak-builder-lint`, since there's a lot happening here. The problem I ran into was probably this one: https://github.com/flatpak/flatpak-builder-tools/issues/381. The lockfile is _indeed_ lockfile version 2 now. So that's, err, _concerning_, but also fortunate it's only the one thing that breaks here :thinking: 

> I don't see any changes there other than the URL update, was this no longer necessary or did they get accidentally left out of the commits?

Yep, I ended up squashing that change into c5126b6108afdc838232bbae6a260ace0c49d138. Mostly this part: https://github.com/flathub/edu.mit.Scratch/pull/55/commits/c5126b6108afdc838232bbae6a260ace0c49d138#diff-1660a6484fcf48528c7f07a34e072c0405e15616909a7ddf1c8df4dfd9189273R99-R107, and the addition of `npm_config_ignore_scripts: 'true'`.

**Comment by wjt** [comment]
bot, build

File: edu.mit.Scratch.yaml [hunk 1]
```diff
@@ -95,16 +95,19 @@ modules:
...
+          - zypak-wrapper ./scratch-desktop "$@"
 
   - name: appdata
     buildsystem: simple
     build-commands:
-      - 'install -Dm 644 edu.mit.Scratch.appdata.xml /app/share/metainfo/edu.mit.Scratch.appdata.xml'
-      - 'install -Dm 644 edu.mit.Scratch.desktop /app/share/applications/edu.mit.Scratch.desktop'
+      - install -Dm 644 edu.mit.Scratch.appdata.xml /app/share/metainfo/edu.mit.Scratch.appdata.xml
+      - install -Dm 644 edu.mit.Scratch.desktop /app/share/applications/edu.mit.Scratch.desktop
+      - install -Dm 644 edu.mit.Scratch.mime.xml /app/share/mime/packages/edu.mit.Scratch.mime.xml
```
Comment (wjt) [inline edu.mit.Scratch.yaml:106]: Interesting, today I learned how mime types are exported by Flatpak apps.

File: edu.mit.Scratch.yaml [hunk 2]
```diff
@@ -85,21 +81,33 @@ modules:
       - type: file
         path: scratch-desktop/node_modules/scratch-gui/package-lock.json
         dest-filename: scratch-gui-package-lock.json
+      - type: archive
+        url: https://downloads.scratch.mit.edu/microbit/scratch-microbit-1.2.0.hex.zip
+        dest: scratch-gui-src-generated
+        sha256: dfd574b709307fe76c44dbb6b0ac8942e7908f4d5c18359fae25fbda3c9f4399
+      - type: inline
+        dest: scratch-gui-src-generated
+        dest-filename: microbit-hex-url.cjs
```
Comment (wjt) [inline edu.mit.Scratch.yaml:90]: I'm sorry to report that this extension doesn't work, but that's not a regression.

**Review by wjt — APPROVED** [review]
I'm just a guy but this looks good to me.

**Pull request merged**